src/components/story-builder/file-operations/SaveDialog.tsx

```diff
@@ -0,0 +1,117 @@
+              <Label htmlFor='title'>Title</Label>
+              <Input
+                id='title'
+                placeholder='Enter session title'
+                value={saveDialog.formData.title}
+                onChange={(e) => handleFieldChange('title', e.target.value)}
+              />
+            </div>
+
+            <div className='space-y-2'>
+              <Label htmlFor='description'>Description</Label>
+              <Textarea
+                id='description'
+                placeholder='Enter session description'
+                value={saveDialog.formData.description}
+                onChange={(e: React.ChangeEvent<HTMLTextAreaElement>) =>
+                  handleFieldChange('description', e.target.value)
+                }
+              />
+            </div>
+
+            <div className='space-y-2'>
+              <Label htmlFor='tags'>Tags</Label>
+              <Input
+                id='tags'
+                placeholder='Enter tags separated by commas'
+                value={saveDialog.formData.tags}
+                onChange={(e) => handleFieldChange('tags', e.target.value)}
+              />
+            </div>
+          </div>
+        )}
+
+        {saveDialog.saveType === 'state' && (
+          <div className='space-y-4'>
+            <div className='space-y-2'>
+              <Label htmlFor='title'>Title</Label>
+              <Input
+                id='title'
+                placeholder='Enter state title'
+                value={saveDialog.formData.title}
+                onChange={(e) => handleFieldChange('title', e.target.value)}
+              />
+            </div>
+
+            <div className='space-y-2'>
+              <Label htmlFor='description'>Description</Label>
+              <Textarea
+                id='description'
+                placeholder='Enter state description'
+                value={saveDialog.formData.description}
+                onChange={(e: React.ChangeEvent<HTMLTextAreaElement>) =>
+                  handleFieldChange('description', e.target.value)
+                }
+              />
+            </div>
+
+            <div className='space-y-2'>
+              <Label htmlFor='tags'>Tags</Label>
+              <Input
+                id='tags'
+                placeholder='Enter tags separated by commas'
+                value={saveDialog.formData.tags}
+                onChange={(e) => handleFieldChange('tags', e.target.value)}
+              />
+            </div>
+          </div>
+        )}
+
+        <div className='flex justify-end space-x-2'>
+          <Button variant='outline' onClick={closeSaveDialog} disabled={saveDialog.isSaving}>
+            Cancel
+          </Button>
+          <Button onClick={performSave} disabled={saveDialog.isSaving || !auth.isAuthenticated}>
+            {saveDialog.isSaving
+              ? 'Saving...'
+              : `${saveDialog.saveType === 'session' ? 'Save Session' : 'Share Story'}`}
```

Comment:
After sharing the story, there should be another modal dialog that contains a link to the story which the user can easily copy.